UPGRADING.md

```diff
@@ -0,0 +1,29 @@
+# Upgrading
+
+Guide to ease migrations between significant changes
+
+## v5 -> v6
+
+Version 6 introduces a number of property and method renames:
```

Comment:
Does it matter if this is included in master before the v6 release?

Reply:
Not particularly, though we could cut some beta releases.